src/glmfit.jl

```diff
@@ -118,6 +119,20 @@ function updateμ!(r::GlmResp{V,D,L}) where {V<:FPVector,D<:Union{Bernoulli,Bino
     end
 end
 
+function updateμ!(r::GlmResp{V,D,L}) where {V<:FPVector,D<:NegativeBinomial,L<:NegativeBinomialLink}
+    y, η, μ, wrkres, wrkwt, dres = r.y, r.eta, r.mu, r.wrkresid, r.wrkwt, r.devresid
+
+    @inbounds for i in eachindex(y, η, μ, wrkres, wrkwt, dres)
+        θ = r.d.r # the shape parameter of the negative binomial distribution
+        μi, dμdη, μomμ = inverselink(L(θ), η[i])
+        μ[i] = μi
+        yi = y[i]
+        wrkres[i] = (yi - μi) / dμdη
+        wrkwt[i] = cancancel(r) ? dμdη : abs2(dμdη) / μomμ
```

Comment:
I think you can skip the `cancel(r)` here because the method specializes both `D` and `L`